opentech/apply/users/management/commands/migrate_users.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+import json
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Group
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from opentech.apply.users.groups import STAFF_GROUP_NAME
+
+
+class Command(BaseCommand):
+    help = "User migration script. Requires a source JSON file."
+    groups = Group.objects.all()
+
+    def add_arguments(self, parser):
+        parser.add_argument('source', nargs='?', type=argparse.FileType('r'), help='Migration source JSON file')
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        with options['source'] as json_data:
+            User = get_user_model()
+            users = json.load(json_data)
+
+            for uid in users:
+                user = users[uid]
+
+                full_name = self.get_full_name(user)
+                user_object, created = User.objects.get_or_create(
+                    email=user.get('mail'),
+                    defaults={
+                        'full_name': full_name,
+                        'drupal_id': uid,
+                    }
+                )
+
+                operation = "Imported" if created else "Processed"
+
+                for group in self.get_user_groups(user):
+                    user_object.groups.add(group)
+
+                # Ensure uid is set
+                user_object.drupal_id = uid
+                user_object.save()
+
+                print(f"{operation} user {uid} ({full_name})")
```

Comment:
> When you are using management commands and wish to provide console output, you should write to self.stdout and self.stderr, instead of printing to stdout and stderr directly. By using these proxies, it becomes much easier to test your custom command. 

https://docs.djangoproject.com/en/2.0/howto/custom-management-commands/#module-django.core.management

Reply:
yep. keep doing this.